python/rmm/_lib/memory_resource.pyx

```diff
@@ -316,27 +231,20 @@ cpdef _initialize(
     global _mr
     _mr = MemoryResource()
 
-    if not pool_allocator:
-        if not managed_memory:
-            typ = CudaMemoryResource
-        else:
-            typ = ManagedMemoryResource
-        args = ()
+    if managed_memory:
+        upstream = ManagedMemoryResource
     else:
-        if not managed_memory:
-            typ = CNMemMemoryResource
-        else:
-            typ = CNMemManagedMemoryResource
+        upstream = CudaMemoryResource
 
+    if pool_allocator:
         if initial_pool_size is None:
             initial_pool_size = 0
 
-        if devices is None:
-            devices = [0]
-        elif isinstance(devices, int):
-            devices = [devices]
-
-        args = (initial_pool_size, devices)
+        typ = PoolMemoryResource
+        args = (upstream(), initial_pool_size)
```

Comment:
Is `PoolMemoryResource` thread safe? Or do we need an additional memory resource to wrap around it to make sure it is thread safe?

Reply:
It is thread safe by default: https://github.com/rapidsai/rmm/pull/443

Reply:
Thanks Ashwin 🙂